tests/core/test_batch_definition.py

```diff
@@ -64,64 +79,15 @@ def test_get_batch_retrieves_only_batch(mocker: pytest_mock.MockFixture):
     mock_asset = mocker.Mock(spec=DataAsset)
     batch_definition.set_data_asset(mock_asset)
 
-    mock_get_batch_list_from_batch_request = mock_asset.get_batch_list_from_batch_request
-
     mock_batch = mocker.Mock(spec=Batch)
-    mock_get_batch_list_from_batch_request.return_value = [mock_batch]
+    mock_asset.get_batch.return_value = mock_batch
 
     # Act
     batch = batch_definition.get_batch()
 
     # Assert
     assert batch == mock_batch
-    mock_get_batch_list_from_batch_request.assert_called_once_with(
-        batch_definition.build_batch_request()
-    )
-
-
-@pytest.mark.unit
-def test_get_batch_retrieves_last_batch(mocker: pytest_mock.MockFixture):
-    # Arrange
-    batch_definition = BatchDefinition[None](name="test_batch_definition")
-    mock_asset = mocker.Mock(spec=DataAsset)
-    batch_definition.set_data_asset(mock_asset)
-
-    mock_get_batch_list_from_batch_request = mock_asset.get_batch_list_from_batch_request
-
-    batch_a = mocker.Mock(spec=Batch)
-    batch_b = mocker.Mock(spec=Batch)
-    batch_list = [batch_a, batch_b]
-    mock_get_batch_list_from_batch_request.return_value = batch_list
-
-    # Act
-    batch = batch_definition.get_batch()
-
-    # Assert
-    assert batch == batch_b
-    mock_get_batch_list_from_batch_request.assert_called_once_with(
-        batch_definition.build_batch_request()
-    )
-
-
-@pytest.mark.unit
-def test_get_batch_raises_error_with_empty_batch_list(mocker: pytest_mock.MockFixture):
```

Comment:
We no longer need this test?

Reply:
The setup here (mocking an empty list returned no longer made sense), but your comment made me realized I should have backfilled tests around this at the asset level. Done in. https://github.com/great-expectations/great_expectations/pull/10295/commits/2774a349510a29500fe437bd75a164d43cc112fe.